src/loaders/stm_load.c

```diff
@@ -270,7 +270,8 @@ static int stm_load(struct module_data *m, HIO_HANDLE * f, const int start)
 	D_(D_INFO "Stored samples: %d", mod->smp);
 
 	for (i = 0; i < mod->ins; i++) {
-		if (mod->xxs[i].len > 1) {
+		if (sfh.ins[i].volume && sfh.ins[i].length) {
```

Comment:
Okay, after some further tests, ST2.3 plays random noise from memory when trying to trigger a sample with a default volume of 0. I first interpreted this noise as being the same as in the file (e.g. with fracture.stm), but when hex-editing some "normal" sample to have a default volume of 0, ST2 plays some other random noise.
As a conclusion, a default volume of 0 is undefined behaviour, so you can actually load those samples or not - if they are used in a pattern, they will sound broken either way.